recon/src/recon.rs

```diff
@@ -429,23 +428,97 @@ where
     }
 }
 
+/// Represents reasons the store is unwilling to persist keys and values
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum InvalidItem<K: Key> {
```

Comment:
After discussion with @nathanielc, we probably don't care about this and can simply return an (optional?) Invalid struct that simply contains counts of each type.